### PR TITLE
PWN-9151 - some more clean up and restore functionality

### PR DESCRIPTION
**Pull request by kabramovp2p**
## Jira Ticket

https://p2pvalidator.atlassian.net/browse/PWN-9151

## Description of Work

PWN-9151 - make 3 modules CryptoModule.kt, MainContainerModule.kt and WalletModule.kt except of HomeModule + add ClaimHandler.kt to move logic from presenter

getHiddenTokensVisibility connect to init visibility state in MyCryptoPresenter.kt

### CODE REVIEW

File: app/src/main/java/org/p2p/wallet/home/ui/crypto/handlers/ClaimHandler.kt [hunk 1]
```diff
@@ -0,0 +1,40 @@
+package org.p2p.wallet.home.ui.crypto.handlers
+
+import org.p2p.core.token.Token
+import org.p2p.wallet.home.model.CryptoPresenterMapper
+import org.p2p.wallet.home.ui.crypto.MyCryptoContract
+import org.p2p.wallet.home.ui.crypto.MyCryptoInteractor
+import org.p2p.wallet.infrastructure.transactionmanager.TransactionManager
+import org.p2p.wallet.transaction.model.TransactionState
+
+class ClaimHandler(
```
Comment (Davran331122) [inline app/src/main/java/org/p2p/wallet/home/ui/crypto/handlers/ClaimHandler.kt:10]: ClaimBundleClickHandler, or ClaimClickListenerDelegate

**Review by Davran331122 — APPROVED** [review]

**Review by eduardmaximovich — APPROVED** [review]

File: app/src/main/java/org/p2p/wallet/home/ui/crypto/MyCryptoPresenter.kt [hunk 2]
```diff
@@ -163,31 +167,9 @@ class MyCryptoPresenter(
...
-                transactionManager.emitTransactionState(
-                    latestActiveBundleId,
-                    TransactionState.ClaimProgress(latestActiveBundleId)
-                )
-                view?.showProgressDialog(
-                    bundleId = bridgeBundle.bundleId,
-                    progressDetails = progressDetails
-                )
-            }
+            claimHandler.handle(view, canBeClaimed, token)
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/home/ui/crypto/MyCryptoPresenter.kt:172]: I would suggest removing view from the params and return here some action or handle it other way 
But it's just a suggestion

File: app/src/main/java/org/p2p/wallet/home/ui/wallet/WalletModule.kt [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+object WalletModule : InjectionModule {
+
+    override fun create() = module {
+        factoryOf(::StrigaKycUiBannerMapper)
+        factoryOf(::WalletPresenterMapper)
+        factoryOf(::StrigaOnRampConfirmedHandler)
+        factoryOf(::StrigaOnRampClickHandler)
+        factoryOf(::StrigaBannerClickHandler)
+        factoryOf(::WalletPresenter) bind WalletContract.Presenter::class
+        factoryOf(::TopUpWalletPresenter) bind TopUpWalletContract.Presenter::class
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/home/ui/wallet/WalletModule.kt:23]: I think top up can be opened from any screen, not only Wallet

File: app/src/main/java/org/p2p/wallet/home/ui/wallet/WalletPresenter.kt [hunk 4]
```diff
@@ -29,7 +27,7 @@ import org.p2p.wallet.utils.unsafeLazy
 
 class WalletPresenter(
     private val usernameInteractor: UsernameInteractor,
-    private val homeMapper: HomePresenterMapper,
+    private val homeMapper: CryptoPresenterMapper,
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/home/ui/wallet/WalletPresenter.kt:30]: Should have its own presenter

**Pull request merged**